do not force expiry for ACK

fkusei opened the pull request:
Having an expiry time is fine for some services, but not for others. The existing implementation did make it impossible to add a non-expiring ACK to a host/service.

fkusei committed:
do not force expiry for ACK

phosei commented:
I agree, but I’d like to keep the current behavior (default duration pre-selected). I suspect that ACKing services without an end date is the exception rather than the rule? How about supporting `no` or `none` so that users can explicitly state their intentions?

fkusei commented:
I don't totally agree with that.

In my usecase, services tend to be broken for a long time due to the nature of the check.
For example, checks that check for updates of installed software. We have to schedule an update to the test system first, this then has to be tested, afterwards we can schedule an update for the staging system, more tests, then production. This sometimes takes multiple months until the actual check is fixed.
Of course, admins can just schedule an expiry in 6 months, but that would defeat the purpose of an expiry.

For me personally, ACKing means "I know it is broken, i will handle the problem. Please do not alert anyone else until this is fixed."

phosei commented:
Well, there’s certainly different ways to look at the semantics of ACKs and downtimes.

If we supported `none`, then users could set `'default_ack_duration': 'none'` in their config. This would solve your problem, wouldn’t it? (We could even think about making `none` the default for this config option.)

fkusei committed:
allow specifying "none" as expiry time

phosei committed:
duration_to_times() returns an Enum now

'False' and 'None' were a bit hard to understand.

phosei committed:
Remove 'force_duration' logic

I don't think this is needed. The only difference now is that it says
"Duration [none]:" when you configure "none" as the default, which is
fine.

phosei committed:
README: Document new defaults

phosei committed:
Setting downtimes makes no sense without a duration

phosei commented:
I’ve pushed a couple more commits, mostly “cleanup”. OK?
